src/elements/content-uploader/ContentUploader.js

```diff
@@ -348,7 +355,14 @@ class ContentUploader extends Component<Props, State> {
                     // webkitRelativePath should be ignored when the upload destination folder is known
                     this.addFilesWithRelativePathToQueue(newFiles, itemUpdateCallback);
                 } else {
-                    this.addFilesWithoutRelativePathToQueue(newFiles, itemUpdateCallback);
+                    this.addFilesWithoutRelativePathToQueue(
+                        newFiles,
+                        this.props.allowPrepopulateFiles
+                            ? () => {
+                                  this.upload();
+                              }
+                            : itemUpdateCallback,
+                    );
```

Comment:
two minor things:
1. you can add `allowPrepopulateFiles` to the destructured object near the top of this method
2. does it work to change?
```
() => {
    this.upload();
}

// to

this.upload
```
```suggestion
                    this.addFilesWithoutRelativePathToQueue(
                        newFiles,
                        allowPrepopulateFiles
                            ? this.upload
                            : itemUpdateCallback,
                    );
```